bson/_cbsonmodule.c

```diff
@@ -2906,6 +2906,26 @@ static PyObject* _cbson_array_of_documents_to_buffer(PyObject* self, PyObject* a
 
     memcpy(&size, string, 4);
     size = BSON_UINT32_FROM_LE(size);
+
+    /* validate the size of the array */
+    if (view.len < (int32_t)size) {
```

Comment:
We also need to account for negative values here: `if (view.len < (int32_t)size || (int32_t)size < BSON_MIN_SIZE)`. Could you add a test case for this?

Reply:
Done